addons/web_editor/static/src/js/editor/odoo-editor/src/OdooEditor.js

```diff
@@ -679,7 +680,7 @@ export class OdooEditor extends EventTarget {
 
     resetContent(value) {
         value = value || '<p><br></p>';
-        this.editable.innerHTML = value;
+        this.editable.innerHTML = xmlToHtml(value);
```

Comment:
AFAIK the native `innerHTML` does not mangle self-closing tags so it is not clear to me: what is the issue here ?

Reply:
I believe we talked about this recently over discord, with the parsing in the back-end not being `method='html'`. And then we roughly traced it back to a jquery update.

Normally the browser can do some good guesswork to fix stuff like `<span/>` being `<span></span>`, but if it gets more complex it can mess up big time. Like the example I gave on both the front-end and back-end commits:
`<t><t/></t><t><t/></t>`
should become
`<t><t></t></t><t><t></t></t>`
but it becomes
`<t><t></t><t><t></t></t>`

You can test the above in your browser, and it's a real usecase where we have 2 `t-if` containing an empty `t-out` each.

It used to be fixed as a side-effect of this modification we applied to jquery, removed in the recent update:
https://github.com/odoo/odoo/commit/9c41ee5091ac06ac3ca71aeac607195c70061e4a#diff-6aa92a9cdf8d2da06543e90a7887ae1275aed41207f9f562ea4c966b9576a041L5796

The mangling is also apparent in 16.0 as part of the 'reset template' feature. If you reset one of the problematic templates in 16.0, it breaks in the same way it does by default now. Because that feature uses the resetContent on the editor which didn't use jquery, which is why I also added the xmlToHtml method there.

---------------------------

The issue can either be fixed this way or by serializing the templates as HTML instead of xml, as in the other commit I made to `base` here.

I only need one of the 2 commits to be merged, I'm just not sure which one is better.